spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -541,8 +546,8 @@ public void abort() throws IOException {
       close();
 
       WriteResult result = delegate.result();
-      cleanFiles(io, Arrays.asList(result.dataFiles()));
-      cleanFiles(io, Arrays.asList(result.deleteFiles()));
+      SparkCleanupUtil.deleteTaskFiles(io, Arrays.asList(result.dataFiles()));
```

Comment:
Don't we want to take advantage of bulk here and delete both data + delete in one call?

Reply:
+1, think it makes sense in case there aren't many data/delete files (for example less than 1000 data and delete files for s3 bulk delete) they can be combined in the same call.

Reply:
Makes sense. Changed.